src/Sylius/Bundle/UiBundle/Renderer/TwigComponentBlockRenderer.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\UiBundle\Renderer;
+
+use Sylius\Bundle\UiBundle\Registry\TemplateBlock;
+use Symfony\UX\TwigComponent\ComponentRendererInterface;
+
+/** @internal */
+final class TwigComponentBlockRenderer implements TemplateBlockRendererInterface
+{
+    public function __construct(
+        private TemplateBlockRendererInterface $decoratedRenderer,
+        private ComponentRendererInterface $componentRenderer,
+    ) {
+    }
+
+    /**
+     * @param array<string, mixed> $context
+     */
+    public function render(TemplateBlock $templateBlock, array $context = []): string
+    {
+        if (null === $templateBlock->getComponent()) {
+            return $this->decoratedRenderer->render($templateBlock, $context);
+        }
+
+        return $this->componentRenderer->createAndRender($templateBlock->getComponent(), ['context' => $templateBlock->getContext() + $context]);
```

Comment:
In this implementation, we do not support context providers while rendering components, which are used only when rendering templates. I'm not sure how much they are used, but I see a certain inconsistency here. 

In general, I wonder if this implementation could be refactored a bit so that you don't decorate another renderer, but decide which one to use according to the type of TemplateBlock, which could also make it easier to create other custom renderers in the future 🤔